swarm/storage/mru/resource.go

```diff
@@ -424,124 +358,81 @@ func (h *Handler) New(ctx context.Context, name string, frequency uint64) (stora
-	rsrc := h.get(nameHash.Hex())
-	if rsrc == nil {
-		return nil, NewError(ErrNothingToReturn, "resource not loaded")
-	}
-	return h.lookup(rsrc, period, 0, refresh, maxLookup)
-}
-
-// Retrieves the latest version of the resource update identified by `name`
-// at the next update block height
-//
+// LookupLatest retrieves the latest version of the resource update with metadata chunk at params.Root
 // It starts at the next period after the current block height, and upon failure
 // tries the corresponding keys of each previous period until one is found
 // (or startBlock is reached, in which case there are no updates).
-//
-// Version iteration is done as in (*Handler).LookupHistorical
-//
-// See also (*Handler).LookupHistorical
-func (h *Handler) LookupLatestByName(ctx context.Context, name string, refresh bool, maxLookup *LookupParams) (*resource, error) {
-	return h.LookupLatest(ctx, ens.EnsNode(name), refresh, maxLookup)
-}
-
-func (h *Handler) LookupLatest(ctx context.Context, nameHash common.Hash, refresh bool, maxLookup *LookupParams) (*resource, error) {
+func (h *Handler) Lookup(ctx context.Context, params *LookupParams) (*resource, error) {
 
-	// get our blockheight at this time and the next block of the update period
-	rsrc := h.get(nameHash.Hex())
+	rsrc := h.get(params.Root.Hex())
 	if rsrc == nil {
 		return nil, NewError(ErrNothingToReturn, "resource not loaded")
 	}
-	currentblock, err := h.getBlock(ctx, rsrc.name)
-	if err != nil {
-		return nil, err
-	}
-	nextperiod, err := getNextPeriod(rsrc.startBlock, currentblock, rsrc.frequency)
-	if err != nil {
-		return nil, err
+	if params.Period == 0 {
+		// get our blockheight at this time and the next block of the update period
+		currentblock, err := h.getBlock(ctx, rsrc.name)
+		if err != nil {
+			return nil, err
+		}
+		var period uint32
+		period, err = getNextPeriod(rsrc.startBlock, currentblock, rsrc.frequency)
+		if err != nil {
+			return nil, err
+		}
+		params.Period = period
 	}
-	return h.lookup(rsrc, nextperiod, 0, refresh, maxLookup)
+	return h.lookup(rsrc, params.Period, params.Version, params.Limit)
 }
 
-// Returns the resource before the one currently loaded in the resource index
-//
+// LookupPreviousByName returns the resource before the one currently loaded in the resource index
 // This is useful where resource updates are used incrementally in contrast to
 // merely replacing content.
-//
 // Requires a synced resource object
-func (h *Handler) LookupPreviousByName(ctx context.Context, name string, maxLookup *LookupParams) (*resource, error) {
-	return h.LookupPrevious(ctx, ens.EnsNode(name), maxLookup)
-}
-
-func (h *Handler) LookupPrevious(ctx context.Context, nameHash common.Hash, maxLookup *LookupParams) (*resource, error) {
-	rsrc := h.get(nameHash.Hex())
+func (h *Handler) LookupPrevious(ctx context.Context, params *LookupParams) (*resource, error) {
```

Comment:
function name mismatch in comment

Reply:
I don't mind fixing this in #704 so I can avoid a potential conflict. Please let me know.

Reply:
This function is gone in #704 as it was not used by any code.

Reply:
I was using this for demonstration purposes in Ljubljana, and I think this functionality should be there. Please put it back.

Reply:
Will do, + fix the comments. Sorry.